modules/Bio/EnsEMBL/Compara/Utils/JIRA.pm

```diff
@@ -22,235 +22,467 @@ package Bio::EnsEMBL::Compara::Utils::JIRA;
 use strict;
 use warnings;
 
-use JSON;
+use Data::Dumper;
 use HTTP::Request;
+use JSON;
 use LWP::UserAgent;
 use Term::ReadKey;
+
+use Bio::EnsEMBL::Utils::Argument qw(rearrange);
+use Bio::EnsEMBL::Utils::IO qw (slurp);
 use Bio::EnsEMBL::Utils::Logger;
 
-use Data::Dumper;
+=head2 new
+
+  Arg[1]      : (optional) string $user - a JIRA username. If not given, uses
+                environment variable $USER as default.
+  Arg[2]      : (optional) string $relco - a Compara RelCo JIRA username. By
+                default, $user.
+  Arg[3]      : (optional) string $division - a Compara division (can be empty
+                for RelCo tickets). If not given, uses environment variable
+                $COMPARA_DIV as default.
+  Arg[4]      : (optional) int $release - Ensembl release version. If not given,
+                uses environment variable $CURR_ENSEMBL_RELEASE as default.
+  Arg[5]      : (optional) string $project - JIRA project name. By default,
+                'ENSCOMPARASW'.
+  Example     : my $jira_adaptor = new Bio::EnsEMBL::Compara::Utils::JIRA('user', 'relco', 'metazoa', 97);
+  Description : Creates a new JIRA object
+  Return type : Bio::EnsEMBL::Compara::Utils::JIRA object
+  Exceptions  : none
 
-sub json_to_jira {
-    my ($self, $json_hash, $issuetype, $parameters, $logger) = @_;
+=cut
 
-    # We can define one or many components
-    my @components;
-    if ($json_hash->{'component'}) {
-        push @components, { 'name' => $json_hash->{'component'} };
+sub new {
+    my $caller = shift;
+    my $class = ref($caller) || $caller;
+    my ( $user, $relco, $division, $release, $project ) = rearrange(
+        [qw(USER RELCO DIVISION RELEASE PROJECT)], @_);
+    my $self = {};
+    bless $self, $class;
+    # Initialize logger
+    $self->{_logger} = Bio::EnsEMBL::Utils::Logger->new(-LOGLEVEL => 'info');
+    # Set username that will be used to create the JIRA tickets
+    if ($user) {
+        $self->{_user} = $self->_validate_username($user);
     } else {
-        push @components, { 'name' => $_ } for @{$json_hash->{'components'}};
+        $self->{_user} = $self->_validate_username($ENV{'USER'});
     }
-    if ($parameters->{'tickets'}->{'component'}) {
-        push @components, { 'name' => $parameters->{'tickets'}->{'component'} };
-    } elsif ( $parameters->{'tickets'}->{'components'} ) {
-        push @components, { 'name' => $_ } for @{$parameters->{'tickets'}->{'components'}};
+    if ($relco) {
```

Comment:
And another perl-ier suggestion ;) :
`$self->{_relco} = $relco ? $self->_validate_username($relco) : $self->{_user};`